File: programmers/src/main/java/shane/leetcode/problems/easy/Q258AddDigits.java
package shane.leetcode.problems.easy;

import org.assertj.core.api.Assertions;
import org.junit.jupiter.api.Test;

public class Q258AddDigits {

    @Test
    void test() {
        Assertions.assertThat(addDigits(38)).isEqualTo(2);
        Assertions.assertThat(addDigits(100)).isEqualTo(1);
        Assertions.assertThat(addDigits(0)).isEqualTo(0);
        Assertions.assertThat(addDigits(10)).isEqualTo(1);
        Assertions.assertThat(addDigits(99)).isEqualTo(9);
    }

    public int addDigits(int num) {
        while (num >= 10) {
            int temp = num;
            num = 0;
            while (temp > 0) {
                num += temp % 10;
                temp /= 10;
            }

        }
        return num;
    }

}

File: src/main/kotlin/shane/leetkode/problems/easy/Q258AddDigits.kt
package shane.leetkode.problems.easy

import org.assertj.core.api.Assertions.*
import org.junit.jupiter.api.Test

class Q258AddDigits {

    @Test
    fun test() {
        assertThat(addDigits(38)).isEqualTo(2)
        assertThat(addDigits(100)).isEqualTo(1)
        assertThat(addDigits(0)).isEqualTo(0)
        assertThat(addDigits(10)).isEqualTo(1)
        assertThat(addDigits(99)).isEqualTo(9)
    }

    fun addDigits(num: Int): Int {
        var num = num
        while (num >= 10) {
            var temp = num
            num = 0
            while (temp > 0) {
                num += temp % 10
                temp /= 10
            }
        }
        return num

    }


}
